src/Microsoft.SqlTools.Credentials/Credentials/OSX/OSXCredentialStore.cs

```diff
@@ -96,10 +96,40 @@ private KeyChainItemHandle LookupKeyChainItem(string credentialId)
                 out passwordPtr,
                 out item);
 
-            if(status == Interop.Security.OSStatus.ErrSecSuccess)
+            if (status == Interop.Security.OSStatus.ErrSecSuccess)
             {
                 return new KeyChainItemHandle(item, passwordPtr, passwordLength);
             }
+            else
+            {
+#pragma warning disable 0612
+                // Intentional fallback to Unicode to retrieve old passwords before encoding shift
+                status = Interop.SecurityOld.SecKeychainFindGenericPassword(
+                    IntPtr.Zero,
+                    InteropUtils.GetLengthInBytes(credentialId),
+                    credentialId,
+                    0,
+                    null,
+                    out passwordLength,
+                    out passwordPtr,
+                    out item);
+#pragma warning restore 0612
+                if (status == Interop.Security.OSStatus.ErrSecSuccess)
+                {
+                    var handle = new KeyChainItemHandle(item, passwordPtr, passwordLength);
+                    // Migrate credential to 'Auto' encoding.
+                    if (handle != null)
+                    {
+                        var saveResult = this.Save(credential: new Credential(credentialId, handle.Password));
+                        if (saveResult)
+                        {
+                            // Safe to delete old password now.
+                            this.DeletePassword(credentialId);
```

Comment:
Isn't this going to delete the new password? This should be calling

```csharp
Interop.Security.OSStatus status = Interop.Security.SecKeychainItemDelete(handle);
```

instead (right now it's doing a new lookup of the credential). You could also make new `LookupKeyChainItemOld` and `DeletePasswordOld` helper methods if you wanted (or probably in that case would be to just add a parameter to the existing ones to specify whether it's using the old implementations)

Reply:
Good catch! Will address it.